databricks/sdk/mixins/sql.py

```diff
@@ -0,0 +1,231 @@
+
+    @staticmethod
+    def _raise_if_needed(status: StatementStatus):
+        if status.state not in [StatementState.FAILED, StatementState.CANCELED, StatementState.CLOSED]:
+            return
+        msg = status.state.value
+        if status.error is not None:
+            msg = f"{msg}: {status.error.error_code.value} {status.error.message}"
+        raise RuntimeError(msg)
+
+    def execute(self,
+                warehouse_id: str,
+                statement: str,
+                *,
+                byte_limit: Optional[int] = None,
+                catalog: Optional[str] = None,
+                schema: Optional[str] = None,
+                timeout: timedelta = timedelta(minutes=20),
+                ) -> ExecuteStatementResponse:
+        """(Experimental) Execute a SQL statement and block until results are ready,
+        including starting the warehouse if needed.
+
+        This is a high-level implementation that works with INLINE result disposition,
+        fetching records in JSON format. It can be considered as a quick way to run SQL
+        queries that return up to a dozen megabytes of results by just depending on
+        Databricks SDK for Python without having to pull hundreds of megabytes of
+        library dependencies of Apache Arrow.
+
+        To seamlessly iterate over the rows from query results, please use `execute_fetch_all()`
+        method on this class.
+
+        :param warehouse_id: str
+          Warehouse upon which to execute a statement.
+        :param statement: str
+          SQL statement to execute
+        :param byte_limit: int (optional)
+          Applies the given byte limit to the statement's result size. Byte counts are based on internal
+          representations and may not match measurable sizes in the JSON format.
+        :param catalog: str (optional)
+          Sets default catalog for statement execution, similar to `USE CATALOG` in SQL.
+        :param schema: str (optional)
+          Sets default schema for statement execution, similar to `USE SCHEMA` in SQL.
+        :param timeout: timedelta (optional)
+          Timeout after which the query is cancelled. If timeout is less than 50 seconds,
+          it is handled on the server side. If the timeout is greater than 50 seconds,
+          Databricks SDK for Python cancels the statement execution and throws `TimeoutError`.
+        :return: ExecuteStatementResponse
+        """
+        # The wait_timeout field must be 0 seconds (disables wait),
+        # or between 5 seconds and 50 seconds.
+        wait_timeout = None
+        if 5 <= timeout.total_seconds() <= 50:
+            # set server-side timeout
+            wait_timeout = f"{timeout.total_seconds()}s"
+
+        _LOG.debug(f"Executing SQL statement: {statement}")
+
+        # technically, we can do Disposition.EXTERNAL_LINKS, but let's push it further away.
+        # format is limited to Format.JSON_ARRAY, but other iterations may include ARROW_STREAM.
```

Comment:
Actually can we use EXTERNAL_LINKS here? It is much more scalable both in terms of concurrency that the service can handle and also in terms of result sizes (max 100GB vs max 25MB). As this codes abstracts away the fetching, it doesn't add complexity for the user.

Reply:
@sander-goos i'll try